### PR TITLE
[PyOV] Drop Python 3.7 in OpenVINO Runtime

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Drop of Python 3.7 in OpenVINO Runtime
 - The other components have separate tickets
 - This PR can serve as a test branch and is expected to be up for a long period of time

### Tickets:
 - 118790
 - 118793

### CODE REVIEW

**Review by mlukasze — APPROVED** [review]

**Comment by akuporos** [comment]
Please, also remove else-statement here: https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/openvino/utils.py#L38

**Comment by ilya-lavrenov** [comment]
We also need to clean up troubleshooting section https://pypi.org/project/openvino/
```
ImportError: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
```

File: docs/install_guides/pypi-openvino-dev.md [hunk 1]
```diff
@@ -170,11 +170,11 @@ alias pip='noglob pip'
 
 On Windows*, some libraries are necessary to run OpenVINO. To resolve this issue, install the [C++ redistributable (.exe)](https://aka.ms/vs/17/release/vc_redist.x64.exe). You can also view a full download list on the [official support page](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist).
 
-### ImportError: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
+### ImportError: libpython3.10m.so.1.0: cannot open shared object file: No such file or directory
```
Comment (ilya-lavrenov) [inline docs/install_guides/pypi-openvino-dev.md:173]: should not we remove this section at all?
At least, `3.10m` is not valid configuration

Comment (p-wysocki) [reply docs/install_guides/pypi-openvino-dev.md]: Good point, I think that this section is not needed as the issue is not OpenVINO related. I removed it from our PyPI pages.

**Review by akuporos — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

**Pull request merged**